pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -220,12 +221,13 @@ public TopicPolicies getTopicPolicies(TopicName topicName,
             NamespaceName namespace = topicName.getNamespaceObject();
             prepareInitPoliciesCache(namespace, new CompletableFuture<>());
         }
-        if (policyCacheInitMap.containsKey(topicName.getNamespaceObject())
-                && !policyCacheInitMap.get(topicName.getNamespaceObject())) {
-            throw new TopicPoliciesCacheNotInitException();
+        synchronized (this) {
```

Comment:
Can we consider relying on `policyCacheInitMap` to avoid `synchronized`?

Reply:
Good idea.